regression/run/references/references.hpp

```diff
@@ -45,7 +45,26 @@ class ArrayWrapper
     double *getArrayConst() const { return arr; }
     const double *getArrayC() { return arr; }
     const double *getArrayConstC() const { return arr; }
-    void fetchArray(double **array, int &isize) { *array = arr; isize = sz; }
+    void fetchArrayPtr(double **array, int *isize) { *array = arr; *isize = sz; }
+    void fetchArrayRef(double *&array, int &isize) { array = arr; isize = sz; }
+    void fetchArrayPtrConst(const double **array, int *isize) {
+        *array = arr;
+        *isize = sz;
+    }
+    void fetchArrayRefConst(const double *&array, int &isize) {
+        array = arr;
+        isize = sz;
+    }
+    void fetchVoidPtr(void **array) { *array = arr; }
+    void fetchVoidRef(void *&array) { array = arr; }
+    bool checkPtr(void *array) { return array == arr ? true : false; }
+    int  sumArray() {
```

Comment:
Thanks for the updates @ltaylor16.

The return type for this function should be `double`, right?

Reply:
Fixed.